src/stores/utils/web3.js

```diff
@@ -4,31 +4,24 @@ import Web3Utils from 'web3-utils'
 const getWeb3 = () => {
   return new Promise(function (resolve, reject) {
     // Wait for loading completion to avoid race conditions with web3 injection timing.
-    window.addEventListener('load', function () {
+    window.addEventListener('load',async function () {
       let web3 = window.web3
+      const { ethereum } = window
 
-      // Checking if Web3 has been injected by the browser (Mist/MetaMask)
-      if (typeof web3 !== 'undefined') {
-        // Use Mist/MetaMask's provider.
+      if (ethereum) {
+        web3 = new window.Web3(ethereum)
+        try {
+          // Request account access
+          await ethereum.enable()
+          processWeb3(web3, resolve,  reject)
+        } catch (error) {
+          console.log(error)
+          const errorMsg = `Metamask account rejected by user`
+          reject({message: errorMsg})
+        }
+      } else if (typeof web3 !== 'undefined') {
         web3 = new window.Web3(web3.currentProvider)
-        web3.version.getNetwork((err, netId) => {
-          const netIdName = getNetworkName(netId)
-          console.log(`This is ${netIdName} network.`, netId)
-          document.title = `${netIdName} - Bridge UI dApp`
-          const defaultAccount = web3.eth.defaultAccount || null;
-          if(defaultAccount === null){
-            reject({message: 'Please unlock your metamask and refresh the page'})
-          }
-          const results = {
-            web3Instance: new Web3(web3.currentProvider),
-            netIdName,
-            netId,
-            injectedWeb3: true,
-            defaultAccount
-          }
-          resolve(results)
-        })
-
+        processWeb3(web3, resolve,  reject)
       } else {
         // Fallback to localhost if no web3 injection.
         const errorMsg = `Metamask is not installed. Please go to
```

Comment:
All the `metamask` references (except the new included in this PRs) were addressed on https://github.com/poanetwork/bridge-ui/pull/138/commits/e5edd4dec3e94834f84346c81a1a1b340e786892 on PR https://github.com/poanetwork/bridge-ui/pull/138. So after that PR is merged, this will be solved.

Reply:
Understood